### ISSUE TITLE
valid-title should allow functions

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [veritem/eslint-plugin-vitest#247]
As of https://github.com/vitest-dev/vitest/issues/3493 -> https://github.com/vitest-dev/vitest/pull/3497, Vitest `describe`s & similar can take in function names. I think the `valid-title` rule should by default no longer complain if a function name is provided.

Before:

```ts
import { describe } from "vitest"

function isBodyWithReplies() { /* ... */ }

describe(isBodyWithReplies, () => {
	//   ~~~~~~~~~~~~~~~~~~ Test title must be a string
	// ...
})
```

After:

```ts
import { describe } from "vitest"

function isBodyWithReplies() { /* ... */ }

describe(isBodyWithReplies, () => {
	//   OK
	// ...
});

describe(1234567890, () => {
	//   ~~~~~~~~~~ Test title must be a function or string
	// ...
});
```

I'd be happy to send a PR for this! ❤️

**Comment by veritem** [comment]
I don't have enough time to fix this but lately. Meanwhile,  a PR would be much appreciated as always.
